tests/remotes/webhdfs.py

```diff
@@ -58,155 +57,7 @@ def read_text(self, encoding=None, errors=None):
-                        yield chunk
-
-                data = read_chunks(
-                    pobj.open("rb", encoding=encoding), chunk_size
-                )
-
-            if progress:
-
-                def reader(_hdfs_path, _progress):
-                    nbytes = 0
-                    for chunk in data:
-                        nbytes += len(chunk)
-                        _progress(_hdfs_path, nbytes)
-                        yield chunk
-                    _progress(_hdfs_path, -1)
-
-                yield reader(hdfs_path, progress)
-            else:
-                yield data
-
-    def walk(self, hdfs_path):
-        import posixpath
-
-        local_path = self._path(hdfs_path)
-        for local_root, dnames, fnames in os.walk(local_path):
-            if local_root == os.fspath(local_path):
-                root = hdfs_path
-            else:
-                root = posixpath.join(
-                    hdfs_path, os.path.relpath(local_root, local_path)
-                )
-            yield (root, dnames, fnames)
-
-    def delete(self, hdfs_path):
-        return self._path(hdfs_path).unlink()
-
-    def status(self, hdfs_path, strict=True):
-        from hdfs.util import HdfsError
-
-        try:
-            return {"length": self._path(hdfs_path).stat().st_size}
-        except FileNotFoundError:
-            if not strict:
-                return None
-            raise HdfsError(
-                f"File does not exist: {hdfs_path}",
-                exception="FileNotFoundException",
-            )
-
-    def checksum(self, hdfs_path):
-        return {
-            "algorithm": "MD5-of-0MD5-of-512CRC32",
-            "bytes": md5md5crc32c(self._path(hdfs_path)) + "00000000",
-            "size": 28,
-        }
-
-    def rename(self, from_path, to_path):
-        from dvc.utils.fs import move
-
-        move(self._path(from_path), self._path(to_path))
-
-    def upload(
-        self,
-        hdfs_path,
-        local_path,
-        chunk_size=2 ** 16,
-        progress=None,
-        **kwargs,
-    ):
-        with open(local_path, "rb") as from_fobj:
-            with self.write(hdfs_path, **kwargs) as to_fobj:
-                nbytes = 0
-                while True:
-                    chunk = from_fobj.read(chunk_size)
-                    if not chunk:
-                        break
-                    if progress:
-                        nbytes += len(chunk)
-                        progress(local_path, nbytes)
-                    to_fobj.write(chunk)
-
-    def download(self, hdfs_path, local_path, **kwargs):
-        from dvc.utils.fs import makedirs
-
-        kwargs.setdefault("chunk_size", 2 ** 16)
-
-        makedirs(os.path.dirname(local_path), exist_ok=True)
-        with open(local_path, "wb") as writer:
-            with self.read(hdfs_path, **kwargs) as reader:
-                for chunk in reader:
-                    writer.write(chunk)
-
-
-@pytest.fixture
-def webhdfs(mocker):
```

Comment:
A bit sad that we no longer have mocked webhdfs, but we could think about mocking with localfs in the future. Or maybe mocking a whole webhdfs server. Not for this PR, of course.